iceoryx_posh/test/moduletests/test_version_info.cpp

```diff
@@ -0,0 +1,254 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/version/version_info.hpp"
+#include "iceoryx_versions.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::version;
+using namespace iox::cxx;
+
+namespace
+{
+class VersionInfoSut : public VersionInfo
+{
+  public:
+    VersionInfoSut(uint16_t versionMajor,
+                   uint16_t versionMinor,
+                   uint16_t versionPatch,
+                   uint16_t versionTweak,
+                   const BuildDateStringType& buildDateString,
+                   const CommitIdStringType& commitIdString) noexcept
+        : VersionInfo(versionMajor, versionMinor, versionPatch, versionTweak, buildDateString, commitIdString){};
+
+    VersionInfoSut(const Serialization& serial) noexcept
+        : VersionInfo(serial){};
+
+    uint16_t getVersionMajor()
+    {
+        return m_versionMajor;
+    }
+    uint16_t getVersionMinor()
+    {
+        return m_versionMinor;
+    }
+    uint16_t getVersionPatch()
+    {
+        return m_versionPatch;
+    }
+    uint16_t getVersionTweak()
+    {
+        return m_versionTweak;
+    }
+
+    VersionInfo::BuildDateStringType getBuildDateString()
+    {
+        return m_buildDateString;
+    }
+
+    VersionInfo::CommitIdStringType getCommitIdString()
+    {
+        return m_commitIdString;
+    }
+};
+} // namespace
+
+class VersionInfo_test : public Test
+{
+  public:
+    VersionInfo_test(){};
+    virtual ~VersionInfo_test(){};
+    virtual void SetUp(){};
+    virtual void TearDown(){};
+};
+
+
+TEST_F(VersionInfo_test, ValidMaxVersionTweak)
+{
+    VersionInfoSut versionInfoSut(65535u, 65535u, 65535u, 65535u, "abc", "efg");
+    EXPECT_EQ(versionInfoSut.getVersionMajor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionMinor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionPatch(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionTweak(), 65535u);
+    EXPECT_EQ(std::string(versionInfoSut.getBuildDateString().c_str()), "abc");
+    EXPECT_EQ(std::string(versionInfoSut.getCommitIdString().c_str()), "efg");
+}
+
+TEST_F(VersionInfo_test, ValidMaxVersionFinalRelease)
+{
+    VersionInfoSut versionInfoSut(65535u, 65535u, 65535u, 0, "", "");
+    EXPECT_EQ(versionInfoSut.getVersionMajor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionMinor(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionPatch(), 65535u);
+    EXPECT_EQ(versionInfoSut.getVersionTweak(), 0u);
+}
+
+TEST_F(VersionInfo_test, SerializationWorking)
+{
+    VersionInfoSut versionInfoSut1(65535u, 2u, 3u, 789u, "abc", "efg");
+    VersionInfoSut versionInfoSut2(versionInfoSut1.operator iox::cxx::Serialization());
```

Comment:
You should not call the conversion operator explicitly. Please write
`VersionInfoSut versionInfoSut2(static_cast<iox::cxx::Serialization>(versionInfoSut1));`